compatibility_lib/compatibility_lib/test_deprecated_dep_finder.py

```diff
@@ -0,0 +1,152 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import mock
+
+from compatibility_lib import deprecated_dep_finder
+from compatibility_lib import fake_compatibility_store
+
+
+class TestDeprecatedDepFinder(unittest.TestCase):
+    PKG_INFO = {
+        'info': {
+            'classifiers': [
+                "Development Status :: 7 - Inactive",
+                "Intended Audience :: Developers",
+                "License :: OSI Approved:: Apache Software License",
+                "Operating System :: POSIX",
+                "Programming Language:: Python :: 2",
+                "Programming Language:: Python :: 2.7",
+                "Programming Language:: Python :: 3",
+                "Programming Language:: Python :: 3.4",
+                "Programming Language:: Python :: 3.5",
+                "Topic :: Internet :: WWW / HTTP",
+            ]
+        }
+    }
+
+    def setUp(self):
+        self.DEP_INFO = {
+            'dep1': {
+                'installed_version': '2.1.0',
+                'installed_version_time': '2018-05-12T16:26:31',
+                'latest_version': '2.1.0',
+                'current_time': '2018-08-27T17:04:57.260105',
+                'latest_version_time': '2018-05-12T16:26:31',
+                'is_latest': True,
+            },
+            'dep2': {
+                'installed_version': '3.6.1',
+                'installed_version_time': '2018-08-13T22:47:09',
+                'latest_version': '3.6.1',
+                'current_time': '2018-08-27T17:04:57.934866',
+                'latest_version_time': '2018-08-13T22:47:09',
+                'is_latest': True,
+            },
+        }
+
+        self.SELF_COMP_RES = ((
+            {
+                'result': 'SUCCESS',
+                'packages': ['package1'],
+                'description': None,
+                'dependency_info': self.DEP_INFO,
+            },
+        ),)
+        self.mock_checker = mock.Mock()
+        self.fake_store = fake_compatibility_store.CompatibilityStore()
+
+        self.mock_checker.get_self_compatibility.return_value = \
+            self.SELF_COMP_RES
+
+        self.patch_checker = mock.patch(
+            'compatibility_lib.deprecated_dep_finder.utils.checker',
+            self.mock_checker)
+        self.patch_store = mock.patch(
+            'compatibility_lib.deprecated_dep_finder.utils.store',
+            self.fake_store)
+
+    def test_constructor_default(self):
+        from compatibility_lib import utils
+
+        with self.patch_checker, self.patch_store:
+            finder = deprecated_dep_finder.DeprecatedDepFinder()
+
+        self.assertEqual(finder.py_version, '3')
+        self.assertTrue(
```

Comment:
assertIsInstance is available after Python 3.2. I am not sure what Python versions should we support, but definitely use that if it is possible.

Reply:
Done.